core/commands/resolve.go

```diff
@@ -72,19 +65,21 @@ Resolve the value of another name recursively:
 		}
 
 		name := req.Arguments()[0]
-		recursive, _, _ := req.Option("recursive").Bool()
-		depth := 1
-		if recursive {
-			depth = namesys.DefaultDepthLimit
+
+		p, err := path.ParsePath(name)
+		if err != nil {
+			res.SetError(err, cmds.ErrClient)
+			return
 		}
 
-		output, err := n.Namesys.ResolveN(req.Context(), name, depth)
+		node, err := core.Resolve(req.Context(), n, p)
```

Comment:
if `core.Resolve` is used here to replace `n.Namesys.ResolveN`, should it be the case in  https://github.com/ipfs/go-ipfs/blob/master/core/commands/ipns.go#L84 too (and https://github.com/ipfs/go-ipfs/blob/master/core/commands/dns.go#L65)?